src/metrics/include/metrics/metrics.hpp

```diff
@@ -82,23 +133,44 @@ class RegistryT final {
     // OperationsMap is a map of
     // actor name -> operation name -> actor id -> OperationImpl (time series).
     using OperationsMap = std::unordered_map<std::string, OperationsByType>;
+    // Map from "Actor.Operation.Phase" to a Collector.
+    using CollectorsMap = std::unordered_map<std::string, v2::Collector>;
 
 public:
     using clock = ClockSource;
 
     explicit RegistryT() = default;
 
+    explicit RegistryT(MetricsFormat format, std::string path_prefix) {
+        _format = std::move(format);
+        _path_prefix = std::move(path_prefix);
+        if (_format.useGrpc()) {
+            boost::filesystem::create_directories(_path_prefix);
+        }
+    }
+
+
     OperationT<ClockSource> operation(std::string actorName,
                                       std::string opName,
                                       ActorId actorId,
                                       std::optional<genny::PhaseNumber> phase = std::nullopt) {
+        std::string name;
+        if (_format.useGrpc()) {
+            name = createName(actorName, opName, phase);
+            _collectors.try_emplace(name, name, _path_prefix);
+        }
         auto& opsByType = this->_ops[actorName];
         auto& opsByThread = opsByType[opName];
-        auto opIt =
-            opsByThread
-                .try_emplace(
-                    actorId, std::move(actorName), *this, std::move(opName), std::move(phase))
-                .first;
+        auto opIt = opsByThread
+                        .try_emplace(actorId,
+                                     actorId,
+                                     std::move(actorName),
+                                     *this,
+                                     std::move(opName),
+                                     std::move(phase),
+                                     _path_prefix,
+                                     name)
```

Comment:
looks like `name` may be default-constructed here?

Reply:
Yeah should only be used in gRPC situations, made it optional to signify that.